// terra-button/lib/terra-dev-site/test/button/ActiveVariantButtons.test.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _IconSquareSymbol = _interopRequireDefault(require("terra-icon/lib/icon/IconSquareSymbol"));

var _bind = _interopRequireDefault(require("classnames/bind"));

var _ButtonModule = _interopRequireDefault(require("terra-button/lib/Button.module.scss"));

var _Button = _interopRequireDefault(require("../../../Button"));

var _ButtonTestCommonModule = _interopRequireDefault(require("./ButtonTestCommon.module.scss"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var cx = _bind.default.bind(_ButtonTestCommonModule.default);

var cy = _bind.default.bind(_ButtonModule.default);

var NeutralButton = function NeutralButton() {
  return /*#__PURE__*/_react.default.createElement("div", {
    id: "neutral",
    className: cx('spacing')
  }, /*#__PURE__*/_react.default.createElement(_Button.default, {
    id: "neutralActive",
    text: "Neutral-Active",
    variant: "neutral",
    className: cy('is-active')
  }), /*#__PURE__*/_react.default.createElement("span", {
    id: "neutralActiveFocusSpan",
    className: cx('spacing')
  }, /*#__PURE__*/_react.default.createElement(_Button.default, {
    id: "neutralActiveFocus",
    text: "Neutral-Active+Focus",
    variant: "neutral",
    className: cy(['is-active', 'is-focused'])
  })));
};

var EmphasisButton = function EmphasisButton() {
  return /*#__PURE__*/_react.default.createElement("div", {
    id: "emphasis",
    className: cx('spacing')
  }, /*#__PURE__*/_react.default.createElement(_Button.default, {
    id: "emphasisActive",
    text: "Emphasis-Active",
    variant: "emphasis",
    className: cy('is-active')
  }), /*#__PURE__*/_react.default.createElement("span", {
    id: "emphasisActiveFocusSpan",
    className: cx('spacing')
  }, /*#__PURE__*/_react.default.createElement(_Button.default, {
    id: "emphasisActiveFocus",
    text: "Emphasis-Active+Focus",
    variant: "emphasis",
    className: cy(['is-active', 'is-focused'])
  })));
};

var GhostButton = function GhostButton() {
  return /*#__PURE__*/_react.default.createElement("div", {
    id: "ghost",
    className: cx('spacing')
  }, /*#__PURE__*/_react.default.createElement(_Button.default, {
    id: "ghostActive",
    text: "Ghost-Active",
    variant: "ghost",
    className: cy('is-active')
  }), /*#__PURE__*/_react.default.createElement("span", {
    id: "ghostActiveFocusSpan",
    className: cx('spacing')
  }, /*#__PURE__*/_react.default.createElement(_Button.default, {
    id: "ghostActiveFocus",
    text: "Ghost-Active+Focus",
    variant: "ghost",
    className: cy(['is-active', 'is-focused'])
  })));
};

var DeemphasisButton = function DeemphasisButton() {
  return /*#__PURE__*/_react.default.createElement("div", {
    id: "deempshasis",
    className: cx('spacing')
  }, /*#__PURE__*/_react.default.createElement(_Button.default, {
    id: "de-emphasisActive",
    text: "De-emphasis-Active",
    variant: "de-emphasis",
    className: cy('is-active')
  }), /*#__PURE__*/_react.default.createElement("span", {
    id: "de-emphasisActiveFocusSpan",
    className: cx('spacing')
  }, /*#__PURE__*/_react.default.createElement(_Button.default, {
    id: "de-emphasisActiveFocus",
    text: "De-emphasis-Active+Focus",
    variant: "de-emphasis",
    className: cy(['is-active', 'is-focused'])
  })));
};

var ActionButton = function ActionButton() {
  return /*#__PURE__*/_react.default.createElement("div", {
    id: "action",
    className: cx('spacing')
  }, /*#__PURE__*/_react.default.createElement(_Button.default, {
    id: "actionActive",
    text: "Action-Active",
    icon: /*#__PURE__*/_react.default.createElement(_IconSquareSymbol.default, null),
    variant: "action",
    className: cy('is-active')
  }), /*#__PURE__*/_react.default.createElement("span", {
    id: "actionActiveFocusSpan",
    className: cx('spacing')
  }, /*#__PURE__*/_react.default.createElement(_Button.default, {
    id: "actionActiveFocus",
    text: "Action-Active+Focus",
    icon: /*#__PURE__*/_react.default.createElement(_IconSquareSymbol.default, null),
    variant: "action",
    className: cy(['is-active', 'is-focused'])
  })));
};

var UtilityButton = function UtilityButton() {
  return /*#__PURE__*/_react.default.createElement("div", {
    id: "utility",
    className: cx('spacing')
  }, /*#__PURE__*/_react.default.createElement(_Button.default, {
    id: "utilityActive",
    text: "Utility-Active",
    icon: /*#__PURE__*/_react.default.createElement(_IconSquareSymbol.default, null),
    variant: "utility",
    className: cy('is-active')
  }), /*#__PURE__*/_react.default.createElement("span", {
    id: "utilityActiveFocusSpan",
    className: cx('spacing')
  }, /*#__PURE__*/_react.default.createElement(_Button.default, {
    id: "utilityButtonActiveFocus",
    text: "Utility-Active+Focus",
    icon: /*#__PURE__*/_react.default.createElement(_IconSquareSymbol.default, null),
    variant: "utility",
    className: cy(['is-active', 'is-focused'])
  })));
};

var activeButtons = function activeButtons() {
  return /*#__PURE__*/_react.default.createElement("div", null, /*#__PURE__*/_react.default.createElement(NeutralButton, null), /*#__PURE__*/_react.default.createElement(EmphasisButton, null), /*#__PURE__*/_react.default.createElement(GhostButton, null), /*#__PURE__*/_react.default.createElement(DeemphasisButton, null), /*#__PURE__*/_react.default.createElement(ActionButton, null), /*#__PURE__*/_react.default.createElement(UtilityButton, null));
};

var _default = activeButtons;
exports.default = _default;
