// terra-button/lib/terra-dev-site/test/button/Variants/UtilityButton.test.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _IconSquareSymbol = _interopRequireDefault(require("terra-icon/lib/icon/IconSquareSymbol"));

var _bind = _interopRequireDefault(require("classnames/bind"));

var _Button = _interopRequireDefault(require("../../../../Button"));

var _ButtonTestCommonModule = _interopRequireDefault(require("../ButtonTestCommon.module.scss"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var cx = _bind.default.bind(_ButtonTestCommonModule.default);

var UtilityButton = function UtilityButton() {
  return /*#__PURE__*/_react.default.createElement("div", {
    id: "utility",
    className: cx('button-wrapper')
  }, /*#__PURE__*/_react.default.createElement(_Button.default, {
    id: "utilityButton",
    text: "Utility",
    icon: /*#__PURE__*/_react.default.createElement(_IconSquareSymbol.default, null),
    variant: "utility"
  }), /*#__PURE__*/_react.default.createElement(_Button.default, {
    id: "utilityButtonDisabled",
    text: "Utility Disabled",
    icon: /*#__PURE__*/_react.default.createElement(_IconSquareSymbol.default, null),
    variant: "utility",
    isDisabled: true
  }));
};

var _default = UtilityButton;
exports.default = _default;
